### PR TITLE
MPP-2551 - Premium Promo Banners UI

### PR DESCRIPTION
**Pull request by codemist**
# New feature description

This adds the UI components for the premium promo banners that will be A/B tested.

I've split these specific banners into their own `.tsx` file to make it easier on Maxx to implement the A/B testing architecture. 

image

# Checklist

- [x] l10n changes have been submitted to the l10n repository, if any.
- [x] All acceptance criteria are met.
- [x] I've added or updated relevant docs in the docs/ directory.
- [x] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: frontend/src/components/dashboard/PremiumPromoBanners.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  profile: ProfileData;
+  runtimeData?: RuntimeData;
+};
+
+export const PremiumPromoBanners = (props: Props) => {
+  const banners: ReactNode[] = [];
+
+  if (
+    !props.profile.has_premium &&
+    isPeriodicalPremiumAvailableInCountry(props.runtimeData)
```
Comment (maxxcrawford) [inline frontend/src/components/dashboard/PremiumPromoBanners.tsx:20]: Suggestion: Can we add a waffle flag to hide these by default? I don't want this to ship with all four of these displayed accidentally. 

I do realize that we may need to ship one of these banners. If that's necessary, check with Mel/Eduardo on which is the preferred one to ship.

File: frontend/src/components/dashboard/PremiumPromoBanners.tsx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  const { l10n } = useLocalization();
+
+  return (
+    <Banner
+      type="promo"
+      title={l10n.getString("banner-ab-premium-promo-stop-spam-headline")}
+      illustration={{
+        img: phoneImage,
+        type: "flex-end",
+      }}
```
Comment (maxxcrawford) [inline frontend/src/components/dashboard/PremiumPromoBanners.tsx:49]: question: Does this need an `alt` tag?

cc/ @Vinnl

**Review by maxxcrawford — APPROVED** [review]
These banners look great! Thanks for separating them out into their own template. 

Overall UX question: Should this banner be dismissible like the bundle upgrade banner is?

File: frontend/src/components/dashboard/PremiumPromoBanners.tsx (hunk 2 above)
Comment (Vinnl) [reply frontend/src/components/dashboard/PremiumPromoBanners.tsx:49]: If this is this image:

image

I don't think it conveys any information, so the empty alt tag (`alt=""`) seems fine to me.

File: frontend/src/components/dashboard/PremiumPromoBanners.tsx (hunk 1 above)
Comment (codemist) [reply frontend/src/components/dashboard/PremiumPromoBanners.tsx:20]: Good idea - I'll ask the rest of the team.

File: frontend/src/pages/accounts/profile.page.tsx [hunk 3]
```diff
@@ -359,6 +360,11 @@ const Profile: NextPage = () => {
 
   const banners = (
     <section className={styles["banners-wrapper"]}>
+      {!profile.has_premium &&
+      isPeriodicalPremiumAvailableInCountry(runtimeData.data) &&
+      isFlagActive(runtimeData.data, "premium_promo_banners") ? (
```
Comment (codemist) [inline frontend/src/pages/accounts/profile.page.tsx:365]: @maxxcrawford I moved the conditions to `profile.page.tsx`, it seems to make more sense to keep track of the logic of where and when it should be displayed.

**Pull request merged**